app/code/core/Mage/Catalog/Model/Product/Image.php

```diff
@@ -183,7 +183,7 @@ protected function _getMemoryLimit()
     {
         $memoryLimit = trim(strtoupper(ini_get('memory_limit')));
 
-        if (!isSet($memoryLimit[0])){
+        if (!isSet($memoryLimit)){
```

Comment:
The value of `isset()` on an empty string is `true`. This should be `if (empty($memoryLimit)) {`.

```
php > $a = '';
php > var_dump(isset($a));
php shell code:1:
bool(true)
```